hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/action/commit/FlinkWriteHelper.java

```diff
@@ -91,13 +91,14 @@ public static FlinkWriteHelper newInstance() {
   @Override
   public List<HoodieRecord<T>> deduplicateRecords(
       List<HoodieRecord<T>> records, HoodieIndex<?, ?> index, int parallelism) {
+    final boolean hasInsert = records.get(0).getCurrentLocation().getInstantTime().equals("I");
```

Comment:
how about renaming this as `isInsertBucket` and add a comment to explain why we need this.

Reply:
The `keyedRecords` can be made more efficient:
```java
Map<Object, List<HoodieRecord<T>>> keyedRecords = records.stream()
        .collect(Collectors.groupingBy(record -> record.getKey().getRecordKey()))
```